isobus/src/isobus_diagnostic_protocol.cpp

```diff
@@ -1227,6 +1044,50 @@ namespace isobus
 		}
 	}
 
+	bool DiagnosticProtocol::parse_j1939_network_states(const CANMessage &message)
+	{
+		bool retVal = false;
+
+		if ((CAN_DATA_LENGTH == message.get_data_length()) &&
+		    (static_cast<std::uint32_t>(CANLibParameterGroupNumber::DiagnosticMessage13) == message.get_identifier().get_parameter_group_number()))
+		{
+			const auto &messageData = message.get_data();
+
+			auto command = static_cast<StopStartCommand>(messageData[0] & (DM13_NETWORK_BITMASK << (DM13_BITS_PER_NETWORK * static_cast<std::uint8_t>(networkType))));
+			switch (command)
+			{
+				case StopStartCommand::StopBroadcast:
+					broadcastState = false;
+					break;
+
+				case StopStartCommand::StartBroadcast:
+					broadcastState = true;
+					break;
+
+				default:
+					break;
+			}
+
+			// Check current data link
+			command = static_cast<StopStartCommand>(messageData[0] & (DM13_NETWORK_BITMASK << (DM13_BITS_PER_NETWORK * static_cast<std::uint8_t>(NetworkType::CurrentDataLink))));
+			switch (command)
+			{
+				case StopStartCommand::StopBroadcast:
+					broadcastState = false;
+					break;
+
+				case StopStartCommand::StartBroadcast:
+					broadcastState = true;
+					break;
+
+				default:
+					break;
+			}
```

Comment:
I'm unsure how and if this would work. I looks like the current data link command would overwrite the command from the networkType above. But I assume StopStartCommand would most of the times be `Reserved` or `DontCareNoAction` in the case of CurrentDataLink?

Reply:
Basically if you receive one with "current data link" set, it always applies to your network. I actually see this one being used in the real world most often, rather than specifying explicitly a network to stop broadcasts on, kind of like saying "if you can hear me, please be quiet" haha. Where it gets strange is if you have a bridge device that is bridging a DM13 from another bus to our bus, in which case the sender of the message should specify explicitly instead if possible.